File: backend/src/main/java/com/erp/service/AIService.java
package com.erp.service;

import com.erp.dto.response.AIChatResponseDto;
import com.erp.dto.response.StructuredLeaveSuggestionDto;

public interface AIService {
    AIChatResponseDto chatWithLeaveAssistant(String prompt);
    StructuredLeaveSuggestionDto parseLeaveRequest(String prompt);
    AIChatResponseDto queryHRAssistant(String query);
    AIChatResponseDto generateEmployeeSummary(Long employeeId);
}

File: frontend/src/services/aiService.js
import api from './api'

export const aiService = {
  chatWithLeaveAssistant: async (prompt) => {
    const response = await api.post('/api/ai/chat', { prompt })
    return response.data
  },

  parseLeaveRequest: async (prompt) => {
    const response = await api.post('/api/ai/generate-leave-request', { prompt })
    return response.data
  },

  queryHRAssistant: async (prompt) => {
    const response = await api.post('/api/ai/hr-query', { prompt })
    return response.data
  },

  generateEmployeeSummary: async (employeeId) => {
    const response = await api.get(`/api/ai/employee-summary/${employeeId}`)
    return response.data
  }
}

export default aiService
